library/ssl_misc.h

```diff
@@ -1576,6 +1576,69 @@ MBEDTLS_CHECK_RETURN_CRITICAL
 int mbedtls_ssl_check_curve( const mbedtls_ssl_context *ssl, mbedtls_ecp_group_id grp_id );
 #endif
 
+/**
+ * \brief Return PSA EC info for the specified TLS ID.
+ *
+ * \param tls_id    The TLS ID to look for
+ * \param family    If the TLD ID is supported, then proper \c psa_ecc_family_t
+ *                  value is returned here. Can be NULL.
+ * \param bits      If the TLD ID is supported, then proper bit size is returned
+ *                  here. Can be NULL.
+ * \return          PSA_SUCCESS if the TLS ID is supported,
+ *                  PSA_ERROR_NOT_SUPPORTED otherwise
+ *
+ * \note            If either \c family or \c bits parameters are NULL, then
+ *                  the corresponding value is not returned.
+ *                  The function can be called with both parameters as NULL
+ *                  simply to check if a specific TLS ID is supported.
+ */
+int mbedtls_ssl_get_psa_curve_info_from_tls_id( uint16_t tls_id,
+                                                psa_ecc_family_t *family,
+                                                size_t* bits );
+
+/**
+ * \brief Return \c mbedtls_ecp_group_id for the specified TLS ID.
+ *
+ * \param tls_id    The TLS ID to look for
+ * \return          Proper \c mbedtls_ecp_group_id if the TLS ID is supported,
+ *                  or MBEDTLS_ECP_DP_NONE otherwise
+ */
+mbedtls_ecp_group_id mbedtls_ssl_get_ecp_group_id_from_tls_id( uint16_t tls_id );
+
+/**
+ * \brief Return TLS ID for the specified \c mbedtls_ecp_group_id.
+ *
+ * \param grp_id    The \c mbedtls_ecp_group_id ID to look for
+ * \return          Proper TLS ID if the \c mbedtls_ecp_group_id is supported,
+ *                  or 0 otherwise
+ */
+uint16_t mbedtls_ssl_get_tls_id_from_ecp_group_id( mbedtls_ecp_group_id grp_id );
+
+/**
+ * \brief Return EC's name for the specified TLS ID.
+ *
+ * \param tls_id    The TLS ID to look for
+ * \return          A pointer to a const string with the proper name. If TLS
+ *                  ID is not suppoted, a NULL pointer is returned insted.
+ */
+const char* mbedtls_ssl_get_curve_name_from_tls_id( uint16_t tls_id );
+
+/* This function transforms an ECC group identifier from
+ * https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8
+ * into a PSA ECC group identifier. */
```

Comment:
While solving this issue, I found that `mbedtls_ecc_group_to_psa` is actually quite similar to the `mbedtls_ssl_get_psa_curve_info_from_tls_id` (and related `tls_id_match_table` structure) that I just added. I know that they are on different "levels" (one is at TLS level, while the other is at a lower PSA level), but shouldn't we try to synthesize this?

At least remove `mbedtls_ecc_group_to_psa` calls from `ssl_tls12_server.c` and `ssl_tls12_client.c`. Wdyt?

Reply:
> At least remove `mbedtls_ecc_group_to_psa` calls from `ssl_tls12_server.c` and `ssl_tls12_client.c`. Wdyt?

Indeed I think it would be conceptually cleaner for TLS code to only convert between TLS ID and "other" identifier when "other" can be either PSA or ECP (that is, only use the conversion functions you introduced in this PR), but not to convert between PSA and ECP, that is between two non-TLS identifiers (that is, quit using `mbedtls_ecc_group_to_psa()`).

Not a blocker here, but now might still be a good time to do it, if you feel like it.

Reply:
I agree with you that it would be cleaner, therefore I'm going to change this as well ;)

Reply:
It should be addressed by [this](https://github.com/Mbed-TLS/mbedtls/pull/6863/commits/2b5d3ded1f6ba1018bf5deebcfec6cb1bd6563a4) commit